profile/src/main/java/com/scottlogic/deg/profile/validators/profile/constraints/atomic/TemporalConstraintValidator.java

```diff
@@ -18,55 +18,76 @@
 
-    public TemporalConstraintValidator(String rule, List<FieldDTO> fields)
-    {
+    public TemporalConstraintValidator(String rule, List<FieldDTO> fields) {
         super(rule, fields);
     }
 
     @Override
-    public final ValidationResult validate(TemporalConstraintDTO dto)
-    {
+    public final ValidationResult validate(TemporalConstraintDTO dto) {
         ValidationResult fieldMustBeValid = fieldMustBeValid(dto);
         if (!fieldMustBeValid.isSuccess) return fieldMustBeValid;
 
-        ValidationResult dateTimeMustBeValid = dateTimeMustBeValid(dto);
-        if (!dateTimeMustBeValid.isSuccess) return dateTimeMustBeValid;
+        FieldType fieldType = fields.stream().filter(f -> f.name.equals(dto.field)).findFirst().get().type.getFieldType();
 
-        return fieldTypeMustMatchValueType(dto, FieldType.DATETIME);
+        ValidationResult validationResult;
+
+        switch (fieldType){
+            case DATETIME:
+                validationResult = dateTimeMustBeValid(dto);
+                break;
+            case TIME:
+                validationResult = timeMustBeValid(dto);
+                break;
+            default:
+                throw new IllegalStateException("Unexpected Temporal constraint type: " + fieldType);
+        }
+
+        if (!validationResult.isSuccess) return validationResult;
+
+        return fieldTypeMustMatchValueType(dto, fieldType);
     }
 
-    private ValidationResult dateTimeMustBeValid(TemporalConstraintDTO dto)
-    {
+    private ValidationResult dateTimeMustBeValid(TemporalConstraintDTO dto) {
         String dateTime = dto.getDate();
-        if (dateTime == null || dateTime.isEmpty())
-        {
+        if (dateTime == null || dateTime.isEmpty()) {
             return ValidationResult.failure("DateTime must be specified" + getErrorInfo(dto));
         }
-        try
-        {
+        try {
             OffsetDateTime offsetDateTime = DateTimeFactory.create(dateTime);
             OffsetDateTime max = DateTimeDefaults.get().max();
             OffsetDateTime min = DateTimeDefaults.get().min();
-            if (offsetDateTime.compareTo(max) > 0 || offsetDateTime.compareTo(min) < 0)
-            {
+            if (offsetDateTime.compareTo(max) > 0 || offsetDateTime.compareTo(min) < 0) {
                 return ValidationResult.failure("Dates must be between 0001-01-01T00:00:00.000Z and 9999-12-31T23:59:59.999Z" + getErrorInfo(dto));
             }
             return ValidationResult.success();
+        } catch (Exception e) {
+            return ValidationResult.failure(e.getMessage() + getErrorInfo(dto));
         }
-        catch (Exception e)
-        {
+    }
+
+    private ValidationResult timeMustBeValid(TemporalConstraintDTO dto) {
+
+        String time = dto.getDate();
+        if (time == null || time.isEmpty()) {
+            return ValidationResult.failure("Time must be specified" + getErrorInfo(dto));
+        }
+        try {
+            HelixTime.create(time);
```

Comment:
This is fine for this PR request. An issue should be raised to remove HelixTime and move its validation here in a similar fashion to how DateTime is validated.